Content.Server/StationEvents/Events/SpaceLawChange.cs

```diff
@@ -0,0 +1,131 @@
+using System.Linq;
```

Comment:
State should not be stored in systems; it needs to be stored in components.

You might also want to look into LocalizedDataset, which would provide an easier way to store the set of laws and to select one.

Reply:
> State should not be stored in systems; it needs to be stored in components.
> 
> You might also want to look into LocalizedDataset, which would provide an easier way to store the set of laws and to select one.

@Tayrtahn I did as you asked, but I ran into a problem. component.SequenceLaws does not save its state between runs. I do not know what to do. I need to store the list between event triggers so that the options are not repeated, but the list is not saved. Could you please tell me how to store the list between runs?

Reply:
> State should not be stored in systems; it needs to be stored in components.
> 
> You might also want to look into LocalizedDataset, which would provide an easier way to store the set of laws and to select one.

I move the selected law to the end of the list and select a random law only from the first half of the list so the laws will not be repeated for as long as possible but the list does not save state between runs...

Reply:
> State should not be stored in systems; it needs to be stored in components.
> 
> You might also want to look into LocalizedDataset, which would provide an easier way to store the set of laws and to select one.

@Tayrtahn I have made all the necessary changes. Now System.Linq is used only to be able to use the Except method in lists, the necessary data is stored in the component and the messages are stored in a localized dataset. Everything works.